neqo-transport/src/send_stream.rs

```diff
@@ -808,6 +820,7 @@ pub struct StreamRecoveryToken {
 }
 
 #[cfg(test)]
+#[allow(clippy::cognitive_complexity)]
```

Comment:
```suggestion
```

If you can't write tests that fit within this limit, then maybe we need to have some helper functions.

In this case, it seems like there are a bunch of independent tests in the ones that you have added.  Separate discrete tests are worth doing.  For instance, you can easily separate out the bit where the slice barrier causes a send to be short.

Reply:
> If you can't write tests that fit within this limit, then maybe we need to have some helper functions.

I think the bar for helper functions in test code has to be higher than regular code. It's ok to be explicit (i.e. not hide steps behind a helper fn) but repetitive in test code to a greater degree than prod code. I don't think this code reaches that bar yet.

Also, this warning is helpful for actual code, but is practically useless for test code. There are no conditionals or branching execution here, it's a straight line. Is the complexity just coming from the use of variables? Not sure but it's seeming not that helpful.

Reply:
The warning catches `assert!()`, which apparently expand into a conditional.

That said, if you have that many asserts and branches in a test, then it is getting hard to understand.  I take a very different view on helper functions, as they help with documentation and sometimes allow you to share some code between tests.

More to the point, we have too many tests that are epics.  We should be able to have multiple smaller tests.

Reply:
I agree, and have opened #504. That said I don't feel this should gate acceptance of this PR.